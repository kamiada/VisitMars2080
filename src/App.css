.App {
  text-align: center;
}

body {
  background-image: url("./assets/images/mars-circle.jpg");
  background-repeat: no-repeat;
  background-size: 100%;
  font-family: 'Libre Baskerville', serif;
}

a {
  text-decoration: none;
  color: #140817; 
  text-align: center;
  margin-top: 20%;
}
a:visited {
  color: #140817; 

}

h1 {
  transition: text-shadow 2s linear;

}

h1:hover {
  color:#fff;
  text-shadow: 0 0 10px #fff,
               0 0 20px #fff, 
               0 0 40px #fff;
}
