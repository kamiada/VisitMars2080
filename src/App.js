import "./App.css";
import { Switcher } from "./components";
import { LanguageContext } from "./components/Switcher/language-context";
import { Route, Routes } from "react-router-dom";
import { FirstScreen, SecondScreen } from "./screens";
import { useState } from "react";

function App() {
  // const { language, setLanguage } = useContext(LanguageContext);
  const [language, setLanguage] = useState("ENG");
  function toggleLanguage() {
    setLanguage((language) => {
      if (language === "ENG") {
        return setLanguage("PL");
      }
      if (language === "PL") {
        return setLanguage("FR");
      }
      if (language === "FR") {
        return setLanguage("ENG");
      }
    });
  }
  return (
    <>
      <LanguageContext.Provider value={{ language, toggleLanguage }}>
        <Switcher onClick={() => setLanguage()} />
        <Routes>
          <Route path="/" element={<FirstScreen />} />
          <Route path="/home" element={<SecondScreen />} />
        </Routes>
      </LanguageContext.Provider>
    </>
  );
}

export default App;
