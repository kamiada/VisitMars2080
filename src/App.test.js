/* eslint-disable testing-library/render-result-naming-convention */
import { render, screen } from "@testing-library/react";
import App from "./App";
import React, { Fragment } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";

// test('renders as expected', () => {
//   const app = render(<App />);
//   expect(app).toMatchSnapshot();

// });

describe("App", () => {
  test("Renders as expected", () => {
    const application = render(
      <Fragment>
        <Router>
          <App />
        </Router>
        ,
      </Fragment>
    );
    expect(application).toMatchSnapshot();
  });
});
