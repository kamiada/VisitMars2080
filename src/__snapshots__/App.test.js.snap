// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ exiyks"
      >
        <button
          class="sc-gsnTZi labjjK"
        >
          ENG
        </button>
      </div>
      <div
        class="sc-eCYdqJ fIxumb"
      >
        <nav
          class="sc-jSMfEi bAkmVg"
        >
          <a
            href="/home"
          >
            <h1
              class="sc-gKXOVf QAEMQ"
            >
              MARS
            </h1>
          </a>
        </nav>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ exiyks"
    >
      <button
        class="sc-gsnTZi labjjK"
      >
        ENG
      </button>
    </div>
    <div
      class="sc-eCYdqJ fIxumb"
    >
      <nav
        class="sc-jSMfEi bAkmVg"
      >
        <a
          href="/home"
        >
          <h1
            class="sc-gKXOVf QAEMQ"
          >
            MARS
          </h1>
        </a>
      </nav>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
